### PR TITLE
enable precompression of certain files at highest level

### PR DESCRIPTION
**Pull request by alcohol**

### CODE REVIEW

File: src/Packagist/WebBundle/Package/SymlinkDumper.php [hunk 1]
```diff
@@ -467,7 +475,17 @@ private function dumpRootFile($file)
...
         }
-        $this->writeFile($file, json_encode($this->rootFile));
+
+        $json = json_encode($this->rootFile);
+        $time = time();
+        $this->writeFile($file, $json, $time);
+
+        $pathgz = $file . '.gz';
+
+        if ($this->compress && !file_exists($pathgz)) {
```
Comment (Seldaek) [inline src/Packagist/WebBundle/Package/SymlinkDumper.php]: I'd remove the pathgz. We should always recompress if we rewrite the file. Avoids a lot of potential fuckups.

File: src/Packagist/WebBundle/Package/SymlinkDumper.php [hunk 2]
```diff
@@ -480,8 +498,17 @@ private function dumpListing($path)
...
+        $time = time();
+
         if (!file_exists($path)) {
-            $this->writeFile($path, $json);
+            $this->writeFile($path, $json, $time);
+        }
+
+        $pathgz = $path . '.gz';
+
+        if ($this->compress && !file_exists($pathgz)) {
```
Comment (Seldaek) [inline src/Packagist/WebBundle/Package/SymlinkDumper.php]: Same here remove check

File: app/config/defaults.yml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 parameters:
     packagist_host: ~
     packagist_metadata_dir: "%kernel.cache_dir%/composer-packages-build"
+    packagist_dumper_compress: 0
```
Comment (stof) [inline app/config/defaults.yml]: should be a boolean, not an integer

Comment (alcohol) [inline app/config/defaults.yml]: Read description of property in class maybe and then let me know if you still feel that way?

Comment (stof) [inline app/config/defaults.yml]: ah sorry, I missed the fact that this was not just enabling the compression but also passed to ``gzencode`` itself

**Pull request merged**